### DISCUSSION TITLE
passwd passwd: no record of root in /etc/shadow, using /etc/passwd: Numerical result out of range Changing password for root

### DISCUSSION THREAD

**Discussion by fraternl** [Freetz-NG/freetz-ng#877] [Allgemein]
```
passwd
passwd: no record of root in /etc/shadow, using /etc/passwd: Numerical result out of range
Changing password for root
New password:

```

I didn't notice it before and thought it had something to do with my tinkering.
But no...  I just created an image after I removed the .config. flashed it and did a factory reset.

When I login for the first time I am forced to change the passwd.
I don't get this message then, but after logging in and invoking the command passwd I get that same message.

The Freetz "Previous generation" didn't have this problem.

Is there a special reason for this error message?

**Comment by fda77** [comment]
Whats the source of this message?

**Comment by fraternl** [comment]
I don't understand the question.
If I give the command passwd,  I will get this message.

I flashed a box with Freetz-NG without any additional packages.
I then did a factory reset and it does the same 

Doesn't it on your boxes?

**Comment by fda77** [comment]
Are there changes between freetz and ng regarding "command passwd"?

**Comment by fraternl** [comment]
It's the busybox passwd command

**/etc/shadow** is a symbolic link to **../var/tmp/shadow**
maybe being it a symbolic link causes the problem?


```
grep "no record of"  ./source/host-tools/busybox-1.36.1/loginutils/passwd.c
                                bb_perror_msg("no record of %s in %s, using %s",

```


```
                if (getspnam_r(pw->pw_name, &spw, buffer, sizeof(buffer), &result) != 0
                 || !result /* no error, but no record found either */
                 || strcmp(result->sp_namp, pw->pw_name) != 0 /* paranoia */
                ) {
                        if (errno != ENOENT) {
                                /* LOGMODE_BOTH */
                                bb_perror_msg("no record of %s in %s, using %s",
                                        name, bb_path_shadow_file,
                                        bb_path_passwd_file);
                        }
                        /* else: /etc/shadow does not exist,
                         * apparently we are on a shadow-less system,
                         * no surprise there */
                } else {
                        pw->pw_passwd = result->sp_pwdp;
                }

```

My old freetz environment is on a virtual machine that I can't start at the moment.

Can you confirm it's doing this for all boxes???

**Comment by fda77** [comment]
I dont know what the
```
 if (getspnam_r(pw->pw_name, &spw, buffer, sizeof(buffer), &result) != 0
```
checks, but its seems not okay what AVM did. For linux user management AVM uses the closed source ctlmgr and its possible that manual by you created users get delete and other silly things

**Comment by fraternl** [comment]
> and its possible that manual by you created users get delete and other silly things

I tested all this on factory resetted fritzbox which ran a vanilla Freetz (deleted the **.config** and set it to International ).
I did this to rule out my own tinkering.

Are you saying you can use **passwd** without getting that error message on your box?

Beside this error message I haven't seen strange behaviour.

**Comment by fda77** [comment]
I'm not testing most time, because a reboot of my modem triggers a punishment of German Telekom which make my crappy internet connection even more crappier by losing some mbit.

I wanted to know if all programs you used have this message
